code/modules/mob/living/silicon/pai/software.dm

```diff
@@ -299,22 +298,71 @@
 
 		if("uninstall")
 			if(href_list["cancel"])
-				uninstall_lock = FALSE
+				uninstallprogress = -1
 			if(subscreen == 1)
 				var/target = href_list["uninstall"]
 				if(software.Find(target))
 					var/cost = available_software[target]
+					uninstallprogress = 0 //when canceled or not uninstalling it sits at -1. needs to be 0 or more for the uninstall loop to proceed.
 					if(uninstall_loop(cost))//uninstall operation must finish before refunding RAM
 						ram+=cost
+						if(screen == software_to_screen_name(target)) //if current screen is our target, send them to the uninstall screen.
+							screen = "uninstall"
+						if(software_to_screen_name(target) == "flashlight")
+							lighted = FALSE
+							card.set_light(0)
+						if(software_to_screen_name(target)=="pps")
+							qdel(pps_device)
+							pps_device = null
+						if(software_to_screen_name(target)=="holomap")
+							qdel(holomap_device)
+							holomap_device = null
+						if(software_to_screen_name(target)=="medicalsupplement")//huds
+							remove_hud_by_type(/datum/visioneffect/medical)
+							medHUD = FALSE
+						if(software_to_screen_name(target)=="securitysupplement")//huds
+							remove_hud_by_type(/datum/visioneffect/security/arrest)
+							remove_hud_by_type(/datum/visioneffect/job)
+							secHUD = FALSE
 						software.Remove(target)
+					uninstallprogress = -1
 				else
 					temp = "Trunk <TT> \"[target]\"</TT> not found."
 
 
 	paiInterface()		 // So we'll just call the update directly rather than doing some default checks
 	return
-
-// MENUS
+//helper function
+/mob/living/silicon/pai/proc/software_to_screen_name(var/target)
```

Comment:
instead of this, why don't we just make the screen name equal to the define?

That seems much less messy and more maintainable, we can eliminate this whole proc too

Reply:
not too familiar with how UI stuff works but the reason I made the proc is for line 309:
I need to check if `screen`  (the current screen) is equal to the `target` software. `screen` holds a string of the current window, but the software define itself already holds a value for ram cost.
how would I do this?

Reply:
Ah, I see your conundrum. Various softwares cost the same amount. Okay, this is fine in that case.